proxyclient/hv.py

```diff
@@ -129,6 +139,13 @@ def map_hook(self, ipa, size, read=None, write=None, **kwargs):
         self.vm_hooks.append((read, write, ipa, kwargs))
         assert self.p.hv_map(ipa, (index << 2) | t, size, 1) >= 0
 
+    def trace_irq(self, device, num, count, flags):
+        if flags & self.IRQTRACE_IRQ:
+            self.interrupt_map[num] = device
+        else:
+            self.interrupt_map.pop(num)
```

Comment:
This needs a guard in case the IRQ was already removed.

Reply:
Was intended to be `pop(num, None)`, missed to add all interrupts to mapping dictionary. both fixed